src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -100,11 +100,11 @@ public String getServiceId() {
   }
 
   /**
-   * @return true if the runtime specified by the user is Java8.
+   * @return true if the runtime read from appengine-web.xml is Java8
    */
   public boolean isJava8() {
     String runtime = getRuntime();
-    return runtime != null && runtime.startsWith("java8");
+    return "java8".equals(runtime) || "java8g".equals(runtime);
```

Comment:
Would it be useful to expose this as a constant/enum?  Especially since there seem to be other valid runtimes (`python27`, etc.)